Document/0x05d-Testing-Data-Storage.md

```diff
@@ -1069,22 +1069,35 @@ setContentView(R.layout.activity_main);
 
 #### Overview
 
-Analyzing the memory can help to identify the root cause of different problems, like for example why an application is crashing, but can also be used to identify sensitive data. This section describes how to check for sensitive data and disclosure of data in general within the process memory.
+Analyzing memory can help developers to identify root causes of different problems, like for example application crashes, etc. However, it can also be used to gain access to sensitive data. This section describes how to check for disclosure of data within the process' memory.
+
+First, you need to identify which sensitive information is stored in memory. Basically, if you have a sensitive asset it's very likely that at some point it is loaded in memory. The objective is to verify that these info has as short exposure as possible. 
 
 To be able to investigate the memory of an application a memory dump needs to be created first or the memory needs to be viewed with real-time updates. This is also already the problem, as the application only stores certain information in memory if certain functions are triggered within the application. Memory investigation can of course be executed randomly in every stage of the application, but it is much more beneficial to understand first what the mobile app is doing and what kind of functionalities it offers and also make a deep dive into the (decompiled) source code before making any memory analysis.
 Once sensitive functions are identified, like decryption of data, the investigation of a memory dump might be beneficial in order to identify sensitive data like a key or the decrypted information itself.
 
 #### Static Analysis
 
-First, you need to identify which sensitive information is stored in memory. Then there are a few checks that must be executed:
+Before looking into the source code, it is beneficial to check documentation (if available) and identify application components so that you get the big picture of where a certain data might be exposed. For example, a sensitive data received from backend does not only exist in the final model object, but also might have multiple copies in the HTTP client, the XML parser, etc. Ideally you want all of these copies to be removed from memory as soon as possible.
```

Comment:
> a data

Data is a bit of a strange word because it is sometimes considered plural and singular. In any case, I would just use "data" here, not "a data".